integration-tests/main-app-test/src/test/java/co/elastic/apm/test/TestAppContainer.java

```diff
@@ -1,214 +0,0 @@
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-import java.util.concurrent.TimeUnit;
-
-import static org.assertj.core.api.Assertions.assertThat;
-
-class TestAppContainer extends GenericContainer<TestAppContainer> {
-
-    private static final Logger log = LoggerFactory.getLogger(TestAppContainer.class);
-
-    private static final String JAVAAGENT_PATH = "/tmp/elastic-apm-agent.jar";
-    private static final String APP_PATH = "/tmp/app.jar";
-    private static final String SECURITY_POLICY = "/tmp/security.policy";
-
-    private boolean appJar;
-    private boolean javaAgent;
-    private final List<String> jvmProperties;
-
-    private String remoteDebugArg = null;
-    private List<String> arguments = new ArrayList<>();
-
-    TestAppContainer(String image) {
-        super(DockerImageName.parse(image));
-        this.jvmProperties = new ArrayList<>();
-    }
-
-    public TestAppContainer withAppJar(Path appJar) {
-        assertThat(appJar).isRegularFile();
-        this.withCopyFileToContainer(MountableFile.forHostPath(appJar), APP_PATH);
-        this.appJar = true;
-        return this;
-    }
-
-    TestAppContainer withJavaAgent(Path agentJar) {
-        assertThat(agentJar).isRegularFile();
-        this.withCopyFileToContainer(MountableFile.forHostPath(agentJar), JAVAAGENT_PATH);
-        this.javaAgent = true;
-        return this;
-    }
-
-    public TestAppContainer withSystemProperty(String key) {
-        return withSystemProperty(key, null);
-    }
-
-    public TestAppContainer withSystemProperty(String key, @Nullable String value) {
-
-        StringBuilder sb = new StringBuilder();
-        sb.append("-D").append(key);
-        if (value != null) {
-            sb.append("=");
-            sb.append(value);
-        }
-        jvmProperties.add(sb.toString());
-
-        return this;
-    }
-
-    public TestAppContainer withSecurityManager() {
-        return withSecurityManager(null);
-    }
-
-    public TestAppContainer withSecurityManager(@Nullable Path policyFile) {
-        withSystemProperty("java.security.manager");
-        if (policyFile != null) {
-            withCopyFileToContainer(MountableFile.forHostPath(policyFile), SECURITY_POLICY);
-            withSystemProperty("java.security.policy", SECURITY_POLICY);
-            log.info("using security policy defined in {}", policyFile.toAbsolutePath());
-            try {
-                Files.readAllLines(policyFile).forEach(log::info);
-            } catch (IOException e) {
-                throw new IllegalStateException(e);
-            }
-        }
-        return this;
-    }
-
-    /**
-     * Configures remote debugging automatically for the JVM running in the container.
-     * On the IDE side, all is required is to add debugger listening for incoming connections on port 5005
-     *
-     * @return this
-     */
-    public TestAppContainer withRemoteDebug() {
```

Comment:
Could you maybe extract this functionality to a superclass, which delegates the actual addition of the remote debug args to the subclass?
I'm thinking of the lambda integration test I work on. We could enable remote debug there via `JAVA_TOOL_OPTS` I think, but this can't use this class.

Reply:
Why a superclass ? We could very probably move this implementation to an independant class so it's easier to reuse, even without testcontainers. Would that work for the use-case you are mentioning with lambda integration test ?

Reply:
> Why a superclass?
I was just thinking because the method currently invokes `withExtraHost`. But a static utility method / class would do the trick aswell.

Reply:
discussed this directly, using `withJvmArgumentsVariable` like in `co.elastic.apm.agent.tomcatlogging.TomcatLoggingIT#testLogShading` with `"JAVA_TOOL_OPTIONS"` argument should work with AWS lambda.